pkg/store/bucket.go

```diff
@@ -974,11 +974,193 @@ func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_Serie
 		set := storepb.MergeSeriesSets(res...)
 		for set.Next() {
 			var series storepb.Series
+			var chks []storepb.AggrChunk
 
 			stats.mergedSeriesCount++
 
 			if req.SkipChunks {
 				series.Labels, _ = set.At()
+
+				if err = srv.Send(storepb.NewSeriesResponse(&series)); err != nil {
+					err = status.Error(codes.Unknown, errors.Wrap(err, "send series response").Error())
+					return
+				}
+			} else {
+				series.Labels, series.Chunks = set.At()
+
+				stats.mergedChunksCount += len(series.Chunks)
+				s.metrics.chunkSizeBytes.Observe(float64(chunksSize(series.Chunks)))
+
+				lblsSize := 0
+				for _, lbl := range series.Labels {
+					lblsSize += lbl.Size()
+				}
+
+				for {
+					series.Chunks, chks = splitFrame(series.Chunks, 1024 * 1024 - lblsSize)
```

Comment:
Nice. 

It would be awesome to have some function helper so we can set this on ALL Store API gRPC components. Maybe even as interceptor? With some instrumentation? :heart:

Reply:
Also we can configure such interceptor by flag to turn on / off 

Example Interceptor: https://github.com/thanos-io/thanos/blob/55cb8ca38b3539381dc6a781e637df15c694e50a/pkg/tracing/grpc.go#L35

Reply:
But even StoreAPI wrapper would do as well